docs/DeveloperGuide.md

```diff
@@ -6,17 +6,146 @@
 
 ## Design & implementation
```

Comment:
Consider separating design and implementation into different section because it's explaining on different aspect of your project.